### ISSUE TITLE
Auto-detect guest's country not functioning as expected

### ISSUE THREAD

**Issue by allynat** [WoollyinWalesIT/jomres#357]
Auto-detect guest's country is set to YES. But on making a new booking, the country is consistently displayed as Germany. I am physically in Thailand.

This is tested further by using VPN to move my ip to other countries and the result is the same.

If I set Default country in booking form to be United Kingdom and set Auto-detect guest's country to NO, then the booking form defaults to United Kingdom.

Issue is that Auto-detect should... auto-detect.

**Comment by allynat** [comment]
To confirm - wordpress- 9.16.1

**Comment by jomres** [comment]
Once a session is created, the user's country is determined on the first call. Therefore, regardless of whether or not the IP number has been changed, the country detected will always be whatever it was found to be the first time the detection was done. I did it this way, the location is cached, so that you guys would be able to use the free version of the geodetection service.

To test it, visit the site. The detected country should be X.
Truncate jomres sessions table.
Change your IP (e.g. through VPN).
Visit the site again. Assuming that the VPN's IP number is a different country, then the country change should be detected.
